ifrs17/Utils/Queries.ipynb

```diff
@@ -399,6 +399,67 @@
             "metadata": {},
             "execution_count": 0,
             "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Data Variables"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> LoadPartitionedDataAsync<T,P>(this IDataSource querySource, Guid partition)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    await querySource.Partition.SetAsync<P>(partition);",
+                "\n    return (await querySource.Query<T>().ToArrayAsync());",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> RelaxedQueryPartitionedDataAsync<T,P>(this IWorkspace workspace, IDataSource dataSource, Guid targetPartition, Guid defaultPartition, string format)",
+                "\n    where T : IPartitioned ",
+                "\n    where P : IPartition",
+                "\n{",
+                "\n    var isRelaxed = (format != ImportFormats.Cashflow && typeof(T).Name == nameof(IfrsVariable)) || ",
+                "\n                    (format == ImportFormats.Cashflow && typeof(T).Name == nameof(RawVariable));",
+                "\n",
+                "\n    var variablesFromWorkspace = await workspace.LoadPartitionedDataAsync<T,P>(targetPartition);",
+                "\n    if(targetPartition == defaultPartition || !isRelaxed || variablesFromWorkspace.Any()) return variablesFromWorkspace; ",
+                "\n",
+                "\n    // For scenario re-calculation",
+                "\n    var variablesFromDataSource = await dataSource.LoadPartitionedDataAsync<T,P>(targetPartition);",
+                "\n    if(variablesFromDataSource.Any()) return variablesFromDataSource;",
+                "\n",
+                "\n    // For scenarios related to parameters solely",
+                "\n    var variablesDefault = await dataSource.LoadPartitionedDataAsync<T,P>(defaultPartition);",
+                "\n    await workspace.Partition.SetAsync<P>(targetPartition);",
```

Comment:
I would reset the partition in the LoadPartitionedDataAsync method. 
In there you can: 
1. GetCurrentPartition
2. set the partition 
3. reset to the initial partition

Reply:
done